ujin/programmers/day08/다트게임.js

```diff
@@ -0,0 +1,26 @@
+function solution(dartResult) {
+  const answer = [];
+  let score = 0;
+  for (let i = 0; i < dartResult.length; i++) {
+    if (dartResult[i] >= 0 && dartResult[i] <= 9) {
+      if (dartResult[i] === "1" && dartResult[i + 1] === "0") {
+        score = 10;
+        i++;
+      } else {
+        score = Number(dartResult[i]);
+      }
+    } else if (dartResult[i] === "S") {
+      answer.push(score);
+    } else if (dartResult[i] === "D") {
+      answer.push(score ** 2);
+    } else if (dartResult[i] === "T") {
+      answer.push(score ** 3);
+    } else if (dartResult[i] === "*") {
+      answer[answer.length - 2] *= 2;
+      answer[answer.length - 1] *= 2;
+    } else if (dartResult[i] === "#") {
+      answer[answer.length - 1] *= -1;
+    }
```

Comment:
위의 설명대로 객체를 생성하고 객체내부의 매서드를 활용해서 반복되는 코드를 줄였습니다.
유진씨의 로직은 그대로 사용했습니다.
```suggestion
    } else {
      check[dartResult[i]](score);
    }
```